### PR TITLE
Ripple #111

### PR DESCRIPTION
**Pull request by MarkBelonogoff**

### CODE REVIEW

**Comment by CaptainAchilles** [comment]
Hey mark,
Your code is looking pretty good, but there are a few things I think which need addressing:

`AvailabilitySelector.vue:94`: `course` could be named better. My understanding of this variable may be incorrect; but it seems it handles the availability distribution of a course. Same with `user`. Something like `p<Availability[]){ default: () => [] }` should be fine.
`AvailabilitySelector.vue:94`: `course` variable should have a defined interface. It is generally bad practice in typescript to use a `any` type. Same with `user`.
`AvailabilityRepository.ts:6` `getCourseAvailability` does not return actually return the `Availability[]` interface. It seems to be returning the following:
```typescript
interface __ {
    course_user: CourseUser,
    day: Day,
    time: Time
}
```
`PeerView.vue:6` `v-on:change` and `@change` are the same thing. It gets confusing when you mix them and as such you should use the shorthand syntax where possible to ensure project consistency. I recommend removing the `shuffleData` call in favour of the functional `changeAvailability`.

`recommendations/models.py:16`: There does not seem to be any time validation. It is generally best practice to be normalizing all time values to `UTC`, then converting them to the locale on the client when they are delivered. This ensures consistency across timezones.

`recommendations/models.py:33`: I see you made an effort to follow the style guide. You are almost spot on; but the `toJSON` serializes the response to a format which is then sent to the JavaScript client, as such; the toJSON methods should have keys which follow the client style guides (`camelCase` in this situation).

---
I know there are a lot of things raised here; but they are all quite small. 🎉🎉

File: client/src/components/peers/PeerView.vue [hunk 1]
```diff
@@ -3,7 +3,10 @@
                class="topPadding">
         <md-layout md-flex="100"
                    class="componentSeparator">
-            <availability-selector @change="shuffleData()"></availability-selector>
+            <availability-selector v-on:change="changeAvailability"
```
Comment (CaptainAchilles) [inline client/src/components/peers/PeerView.vue]: `@change` and `v-on:change` are the same thing. Use `@` where possible,

Comment (MarkBelonogoff) [reply client/src/components/peers/PeerView.vue]: Thanks @CaptainAchilles,
I'll work on fixing them tonight.

File: client/src/repositories/AvailabilityRepository.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                "Content-Type": "application/json"
+            }),
+            body: JSON.stringify({
+                day: day,
+                time: time
+            })
+        })
+            .then(x => {
+                return x.json()
+                    .catch(_ => x);
```
Comment (CaptainAchilles) [inline client/src/repositories/AvailabilityRepository.ts]: You should be re-throwing the exception. Silently handling errors is never a good idea

File: server/src/recommendations/models.py [hunk 3]
```diff
@@ -1,6 +1,36 @@
...
+
+class Availability(models.Model):
+    course_user = models.ForeignKey(CourseUser)
+    day = models.ForeignKey(Day)
+    time = models.ForeignKey(Time)
 
-# Create your models here.
+    def toJSON(self):
+        return {
+            "course_user": self.course_user.toJSON(),
```
Comment (CaptainAchilles) [inline server/src/recommendations/models.py]: `course_user` should be removed to be compliant with the client model definition of `Availability `

File: server/src/recommendations/services/AvailabilityService.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    except ObjectDoesNotExist:
+        exists = False
+
+    if exists:
+        availability.delete()
+        return True
+
+    if not exists:
+        day = Day.objects.get(pk=day_id)
+        time = Time.objects.get(pk=time_id)
```
Comment (CaptainAchilles) [inline server/src/recommendations/services/AvailabilityService.py]: You have an unhanded `Day.ObjectDoesNotExist` and `Time.ObjectDoesNotExists` exceptions here.

**Comment by MarkBelonogoff** [comment]
`recommendations/models.py:16`: For the time validation, did you want the following:

1.  "Time" table in the database to stores the entire 24 hour period as utc 
2. A user updates their availability via the front end time, which is displayed in their local time
3. The local time is converted back to utc before being stored

**Comment by CaptainAchilles** [comment]
Yeah - that sounds correct.

**Pull request merged**